src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -309,17 +309,25 @@ void GcsActorManager::HandleGetActorInfo(rpc::GetActorInfoRequest request,
                  << ", job id = " << actor_id.JobId() << ", actor id = " << actor_id;
 
   const auto &registered_actor_iter = registered_actors_.find(actor_id);
+  std::shared_ptr<GcsActor> *ptr = nullptr;
+  auto arena = reply->GetArena();
+  RAY_CHECK(arena != nullptr);
   if (registered_actor_iter != registered_actors_.end()) {
-    reply->unsafe_arena_set_allocated_actor_table_data(
-        registered_actor_iter->second->GetMutableActorTableData());
+    ptr = google::protobuf::Arena::Create<std::shared_ptr<GcsActor>>(
```

Comment:
is the shared_ptr or the actual data being created on the Arena though?

Reply:
the shared_ptr